static/js/compose_actions.js

```diff
@@ -505,7 +506,12 @@ export function quote_and_reply(opts) {
         compose_ui.replace_syntax("[Quoting…]", content, textarea);
         compose_ui.autosize_textarea($("#compose-textarea"));
         // Update textarea caret to point to the new line after quoted message.
-        textarea.caret(prev_caret + content.length + 1);
+        if (composebox_was_open) {
+            // If composebox was closed, `textbox replace`, by default, places
+            // cursor at end of content. So, we don't need to restore
+            // cursor position.
+            textarea.caret(prev_caret + content.length + 1);
+        }
```

Comment:
I think after reading this, the actual bug is that `const prev_caret = textarea.caret();` should be inside the `replace_content` function.  The logic looks correct without the change in this pull request; the problem is that we're incorrectly fetching the caret a while before we use it.

Reply:
The logic here is correct if you think about `open compose quote insertion`. For closed compose, we actually need `content = textbox.val()` instead of just quoted text, which is what `textbox replace` does by default, so we don't need to meddle with cursor position here.

Doing what you are saying is not solving anything.